### PR TITLE
Remove references to Tag where possible

### PR DESCRIPTION
**Pull request by kbrock**
Test only changes.

Tag is based upon classifications, so just use classifications where possible.
Mostly replaced the `Tag.find()` with `Classification.find_by(:tag_id)`

@miq-bot add_label test

### CODE REVIEW

**Comment by kbrock** [comment]
These are test only changes, I would still like to decease our reliance on tags.

**Comment by jrafanie** [comment]
@kbrock is this still good to go?  Maybe rebase and ping Alberto when ready?  I can't merge here

**Comment by kbrock** [comment]
re cops: I used the variables that were in the existing code. I do not want to change them

**Comment by kbrock** [comment]
@jrafanie this is good to go. it only changes the spec by using the `Category` specs. We really should only be using the classifications factory since that creates tags in the proper way.

File: spec/requests/tags_spec.rb [hunk 1]
```diff
@@ -210,36 +210,37 @@
...
-        end.not_to change(Tag, :count)
+        end.not_to change(Classification, :count)
 
         expect(response).to have_http_status(:forbidden)
       end
 
       it "cannot update a tag" do
         api_basic_authorize
-        tag = Tag.create(:name => "Old name")
+        classification = Classification.is_category.create(:name => "test_tag", :description => "Test Tag")
```
Comment (jrafanie) [inline spec/requests/tags_spec.rb]: How is this different from `FactoryBot.create(:classification, :name => "test_tag", :description => "Test Tag")` ?

File: spec/requests/tags_spec.rb [hunk 2]
```diff
@@ -210,36 +210,37 @@
...
+          post api_tag_url(nil, tag), :params => gen_request(:edit, :name => "new_name")
+        end.not_to change { classification.reload.name }
 
         expect(response).to have_http_status(:forbidden)
       end
 
       it "cannot delete a tag through POST" do
         api_basic_authorize
-        tag = Tag.create(:name => "Test tag")
+        tag = Classification.is_category.create(:name => "test_tag", :description => "Test Tag").tag
```
Comment (jrafanie) [inline spec/requests/tags_spec.rb]: Same question as above for all of these `is_category` calls

**Comment by jrafanie** [comment]
I had only a few comments, looks good other than those

**Comment by lpichler** [comment]
@kbrock is CI failure relevant to PR ?

**Pull request merged**